daisy/step_resize_disks.go

```diff
@@ -0,0 +1,98 @@
+//  Copyright 2017 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisy
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"google.golang.org/api/compute/v1"
+)
+
+// ResizeDisks is a Daisy ResizeDisks workflow step.
+type ResizeDisks []*resizeDisk
+
+type resizeDisk struct {
+	Disk
```

Comment:
resizeDisk should be public, also just take disk name and DisksResizeRequest, (something like you have with DetachDisks):

type ResizeDisk struct {
Disk string

DisksResizeRequest *compute.DisksResizeRequest
}

validate should then use regUse

It's a bit wordy and complicated but allows for the API to add new fields to DisksResizeRequest and daisy just works with them.

Reply:
oh, I missed that one. Sorry. Just a sec...

Reply:
I chose to not add the DiskResizeRequest as member but rather to inherit from it, ok? By doing that, the definition of the resize disk step seems more pleasant to see:
```json
"resizes": {
  "ResizeDisks": [
    {
      "Name": "disk1",
      "SizeGb": "128"
    }
  ]
}
```